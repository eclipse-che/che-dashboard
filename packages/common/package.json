{
  "name": "@eclipse-che/common",
  "version": "7.34.0-SNAPSHOT",
  "repository": "https://github.com/eclipse-che/che-dashboard",
  "license": "EPL-2.0",
  "private": true,
  "main": "lib/index.js",
  "engines": {
    "yarn": ">=1.20.0",
    "node": ">=12"
  },
  "scripts": {
    "build": "yarn clean && tsc -b",
    "watch": "yarn build && tsc -w",
    "clean": "rimraf lib tsconfig.tsbuildinfo"
  },
  "devDependencies": {
    "axios": "^0.21.1",
    "rimraf": "^3.0.2",
    "typescript": "~4.4.2"
  }
}
