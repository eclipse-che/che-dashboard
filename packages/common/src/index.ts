/*
 * Copyright (c) 2018-2023 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import helpers from './helpers';
import * as api from './dto/api';

export * from './dto/cluster-info';
export * from './dto/cluster-config';

export { helpers, api };

export const FACTORY_LINK_ATTR = 'factoryLink';
export const ERROR_CODE_ATTR = 'error_code';
export const KUBECONFIG_MOUNT_PATH = '/tmp/.kube';

const common = {
  helpers,
  api,
};
export default common;
