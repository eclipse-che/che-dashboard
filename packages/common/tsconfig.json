{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib",
    "rootDir": "src",
    "composite": true
  },
  "include": [
    "src",
  ],
  "exclude": [
    "node_modules",
    "lib",
  ]
}
