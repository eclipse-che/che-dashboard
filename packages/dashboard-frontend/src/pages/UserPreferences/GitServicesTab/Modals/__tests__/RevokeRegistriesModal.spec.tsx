/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { api } from '@eclipse-che/common';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import RevokeRegistriesModal from '@/pages/UserPreferences/GitServicesTab/Modals/RevokeGitServicesModal';

describe('Revoke Registries Modal', () => {
  const mockOnRevoke = jest.fn();
  const mockOnCancel = jest.fn();

  function getComponent(
    isRevokeModalOpen: boolean,
    selectedItems: api.GitOauthProvider[],
    gitOauth?: api.GitOauthProvider,
  ): React.ReactElement {
    return (
      <RevokeRegistriesModal
        selectedItems={selectedItems}
        onCancel={mockOnCancel}
        onRevoke={mockOnRevoke}
        isOpen={isRevokeModalOpen}
        gitOauth={gitOauth}
      />
    );
  }

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should correctly render the component', () => {
    const component = getComponent(true, [], 'gitlab');
    render(component);

    const text = screen.queryByText("Would you like to revoke git service 'GitLab'?");
    expect(text).toBeTruthy();

    const checkbox = screen.queryByTestId('warning-info-checkbox');
    expect(checkbox).toBeTruthy();

    const deleteButton = screen.queryByTestId('revoke-button');
    expect(deleteButton).toBeTruthy();

    const cancelButton = screen.queryByTestId('cancel-button');
    expect(cancelButton).toBeTruthy();
  });

  it('should correctly render the component with one selected git service', () => {
    const component = getComponent(true, ['github']);
    render(component);

    const label = screen.queryByText("Would you like to revoke git service 'GitHub'?");
    expect(label).toBeTruthy();
  });

  it('should correctly render the component with two selected git services', () => {
    const component = getComponent(true, ['github', 'gitlab']);
    render(component);

    const label = screen.queryByText('Would you like to revoke 2 git services?');
    expect(label).toBeTruthy();
  });

  it('should fire onRevoke event', () => {
    const component = getComponent(true, ['github']);
    render(component);

    const revokeButton = screen.getByTestId('revoke-button');
    expect(revokeButton).toBeDisabled();

    const checkbox = screen.getByTestId('warning-info-checkbox');
    userEvent.click(checkbox);

    expect(revokeButton).toBeEnabled();

    userEvent.click(revokeButton);

    expect(mockOnRevoke).toHaveBeenCalledWith();
  });

  it('should fire onCancel event', () => {
    const component = getComponent(true, ['github', 'gitlab']);
    render(component);

    const deleteButton = screen.getByTestId('revoke-button');
    expect(deleteButton).toBeDisabled();

    const cancelButton = screen.getByTestId('cancel-button');
    userEvent.click(cancelButton);

    expect(mockOnCancel).toBeCalled();
  });
});
