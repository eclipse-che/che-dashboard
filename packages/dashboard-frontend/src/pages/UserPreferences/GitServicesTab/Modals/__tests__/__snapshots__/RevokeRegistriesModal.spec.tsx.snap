// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Revoke Registries Modal should correctly render the closed modal component 1`] = `null`;
